### PR TITLE
feat: Add support for `istio-csr` addon

### LINKED ISSUE #1055: Add istio-csr addon
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

#### What is the outcome that you are trying to reach?

When using Istio mTLS, administrators should be using valid certificates. Cert-manager has an intergration with Istio to sign, deliver and renew certificates used for mTLS. 

#### Describe the solution you would like

Include the istio-csr addon in the `kubernetes-addons` section of this blueprint.

#### Additional context

https://github.com/cert-manager/istio-csr
https://cert-manager.io/docs/projects/istio-csr/

### PR DESCRIPTION
**Pull request by Pacobart**
### What does this PR do?

Adds istio-csr addon. This addon allows istio to use cert-manager for mTLS certificate generation and management.

### Motivation

- Resolves #1055 

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [x] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs require a new example and/or doc page. In general:
- Use an existing example when possible to demonstrate a new addons usage
- A new docs page under `docs/add-ons/*` is required for new a new addon

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: modules/kubernetes-addons/README.md [hunk 1]
```diff
@@ -312,3 +315,315 @@
 
 No outputs.
 <!-- END OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
+
+<!-- BEGIN_TF_DOCS -->
```
Comment (bryantbiggs) [inline modules/kubernetes-addons/README.md:319]: please remove all of this - terraform docs are already available above

File: modules/kubernetes-addons/cert-manager-istio-csr/versions.tf [hunk 2]
```diff
@@ -0,0 +1,18 @@
+terraform {
+  required_version = ">= 1.0.0"
+
+  required_providers {
```
Comment (bryantbiggs) [inline modules/kubernetes-addons/cert-manager-istio-csr/versions.tf:4]: There are no resources used in this addon so all of the required providers can be removed and just leave the required terraform version

File: modules/kubernetes-addons/cert-manager-istio-csr/locals.tf [hunk 3]
```diff
@@ -0,0 +1,22 @@
+locals {
```
Comment (bryantbiggs) [inline modules/kubernetes-addons/cert-manager-istio-csr/locals.tf:1]: We don't need to use any locals here - everything can be defined directly within the helm addon module. you can use https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/modules/kubernetes-addons/chaos-mesh/main.tf as a reference

**Review by bryantbiggs — APPROVED** [review]

**Comment by deepakjeena** [comment]
@Pacobart : Kindly look into : https://github.com/aws-ia/terraform-aws-eks-blueprints/issues/1280

**Pull request merged**